### DISCUSSION TITLE
Logical Properties instead of Directions

### DISCUSSION THREAD

**Discussion by bennypowers** [RedHat-UX/red-hat-design-system#389] [General]
**TL;DR:** We should use logical properties instead of directions everywhere, both in our internal CSS, and in our element APIs

- `left` -> `inline-start`
- `top` -> `block-start`
- `right` -> `inline-end`
- `bottom` -> `block-end`

## What are logical properties?

Logical properties allow us to support RTL and top-to-bottom languages without much fuss.

[Read about CSS logical Properties on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties)

## Who needs to be aware of this

@RedHat-UX/developers and @RedHat-UX/designers alike need to be aware of these properties, and use them in 
1. Designs - XD files and component specs should use logical properties instead of directions. Designers may add directions in parentheses:
  > a highlight appears at the inline-start (left) of the element
2. Element APIs
  ```js
  @property() highlight?: 'inline-start'|'inline-end';
  ```
3. CSS - most importantly, we should actually use the css logical properties in our component styles
  ```css
  border-start-start-radius: 2px;
  margin-block-start: 0;
  ```

**Comment by coreyvickery** [comment]
@bennypowers Can these be tokens or no?

**Reply by bennypowers** [reply]
short answer: no

longer answer: i believe the tokens spec might allow us to package up rules or properties (as opposed to values) as tokens, but we don't have a compelling means of applying those right now. Instead, we have to rely on developer discipline and code review

```css
/* DON'T */
margin-left: auto;
/* DO */
margin-inline-start: auto;
```

**Comment by wesruv** [comment]
Here here! I just learned about these and I 100% agree.

**Comment by coreyvickery** [comment]
@bennypowers Is this something that I or we need to add to the list of design tokens?

**Reply by bennypowers** [reply]
designers should avoid using directions when defining tokens or writing guidelines
